testfixtures/comparison.py

```diff
@@ -685,3 +685,27 @@ def diff(x, y, x_label='', y_label=''):
             y_label or 'second',
             lineterm='')
     )
+
+
+class RangeComparison:
+    """
+    An object that can be used in comparisons of numerics to validate
+    actual falls within a certain range
+    """
+    def __init__(self, lower_bound, upper_bound):
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+
+    def __eq__(self, other):
+        try:
+            0 + other
```

Comment:
@asqui - comparison against Number is totally wrong. What about a date or a datetime?

Without the type check or exception catching-and-raise, this will return False is Python 2 and a TypeError in Python 3, which matches the behaviour or other less-than-or-greater-than comparisons.